File: easy/e326/python3/solution_test.py
import solution


def test() -> str:
    s = solution.Solution()

    input1 = [27, 0, -1, 9, 1]

    expectations = [True, False, False, True, True]

    if len(input1) != len(expectations):
        return "test is invalid"

    for i in range(len(expectations)):
        assert (-2**31 <= input1[i] and input1[i] <= 2**31-1)

        result = s.isPowerOfThree(input1[i])

        failMsg = (f"TEST #{i} FAILED: bad result for {input1[i]}: got {
                   result} expected {expectations[i]}")

        if result != expectations[i]:
            return failMsg

    return "PASS"


if __name__ == '__main__':
    print(test())

File: easy/e1791/cpp/solution_test.cpp
#include "solution.h"

#include <cassert>
#include <iostream>
#include <vector>

int main() {
    std::vector<std::vector<std::vector<int>>> input1 = { {{1,2}, {1,3}}, {{1,2},{2,3},{4,2}}, {{1,2},{5,1},{1,3},{1,4}} };

    std::vector<int> expectations = { 1, 2, 1 };

    if (input1.size() != expectations.size()) {
        std::cerr << "test is invalid";
        return 0;
    }

    Solution s;
    for (int i = 0; i < input1.size(); i++) {
        assert(2 <= input1[i].size() && input1[i].size() <= pow(10, 5) - 1);

        std::vector<std::vector<int>> origInput;

        for (int j = 0; j < input1[i].size(); j++) {
            assert(input1[i][j].size() == 2);
            assert(input1[i][j][0] != input1[i][j][1]);

            origInput.push_back(std::vector<int>{ input1[i][j][0], input1[i][j][1] });
        }

        int result = s.findCenter(input1[i]);

        if (result != expectations[i]) {
            std::cerr << "FAIL: bad result for [";

            for (int j = 0; j < origInput.size(); j++) {
                if (0 < j) {
                    std::cerr << ", ";
                }

                std::cerr << "[" << origInput[j][0];
                std::cerr << ", " << origInput[j][1];
                std::cerr << "]";
            }

            std::cerr << "]: got " << result;
            std::cerr << " expected " << expectations[i];
            std::cerr << std::endl;

            return 0;
        }
    }

    std::cout << "PASS";

    return 0;
}